test/onnx/onnx_test.cpp

```diff
@@ -292,16 +292,24 @@ TEST_CASE(averagepool_3d_test)
 
 TEST_CASE(averagepool_dyn_test)
 {
+    // Pooling with dynamic input and auto padding
     migraphx::program p;
     auto* mm = p.get_main_module();
     auto l0  = mm->add_parameter(
         "0", {migraphx::shape::float_type, {{1, 4}, {3, 3}, {5, 5}, {5, 5}, {5, 5}}});
-    auto ret = mm->add_instruction(migraphx::make_op("pooling",
-                                                     {{"mode", migraphx::op::pooling_mode::average},
-                                                      {"padding", {0, 0, 0, 0, 0, 0}},
-                                                      {"stride", {1, 1, 1}},
-                                                      {"lengths", {3, 3, 3}}}),
-                                   l0);
+    auto ret = mm->add_instruction(
+        migraphx::make_op(
+            "pooling",
+            {
+                {"mode", migraphx::op::pooling_mode::average},
+                {"stride", {2, 2, 2}},
+                {"lengths", {3, 3, 3}},
+                // TODO: explicit padding is not used in auto padding scenario, but it won't
+                // default to the value the parser adds (all 0's vs all 1's).  Is this a problem?
+                {"padding", {1, 1, 1, 1, 1, 1}},
+                {"padding_mode", 2},
+            }),
+        l0);
```

Comment:
Keep the old test that did dynamic shape average pool. Create a new one that does dynamic shape auto padding.
The padding set to ones problem is from my common above in parse_pooling.cpp L187.

Reply:
New test added.  Onnx parsing does not allow both the auto_pad attribute and a specified set of padding values, although the runtime code would accept it and then ignore the padding inputs.  There's also a potential inconsistency with the intent of the Onnx attribute:  auto_pad is deprecated in the Onnx spec, but internally it's used as the way we get the output size to follow the rule found in `calculate_padding` when input size is dynamic:

`int64_t output_dim     = (input_dim + stride - 1) / stride;`

I think the new test gets it right.